File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        int n = nums.length;
        long totalSum = 0;

        for(int x : nums){
            totalSum += x;
        }

        long leftSum = 0;
        int count = 0;

        for(int i = 0; i < n-1; i++){
            leftSum += nums[i];

            long rightSum = totalSum - leftSum;

            if(leftSum >= rightSum){
                count++;
            }
        }


        return count;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = nums.size();
        long leftSum = 0, totalSum = 0;
        for(int i = 0; i < n; i++){
            totalSum += nums[i];
        }


        int count = 0;
        for(int i =0; i < n-1; i++){
            leftSum += nums[i];

            long rightSum = totalSum-leftSum;

            if(leftSum >= rightSum){
                count++;
            }
        }

        return count;
    }
};

/*
10 + 4 -8 +7

10, 14, 6, 13

*/